src/include/distributed/multi_client_executor.h

```diff
@@ -72,6 +75,28 @@ typedef enum
 } BatchQueryStatus;
 
 
+/* Enumeration to track what the task associated with a connection is waiting for */
+typedef enum WaitStatus
+{
+	WAIT_BUSY, /* no IO required, ready to be processed further */
+	WAIT_ERROR, /* error occured */
+	WAIT_READ_READY, /* waiting for connection to become ready for reads */
+	WAIT_WRITE_READY /* waiting for connection to become ready for writes */
+} WaitStatus;
+
+
+struct pollfd; /* forward declared, to avoid having to include poll.h */
+
+
+typedef struct WaitInfo
+{
+	int allocated;
```

Comment:
Isn't this the length of the `pollfds` array? For convention's sake can we move it to be adjacent to that and rename it to be more clear (`pollfdcount`?). Even though the array isn't inline in the `struct`, I'm still used to seeing lengths next to the end-of-struct arrays.